nodejs/node
# tools: added support for notarytool for osx notarization

### Main Changes

- Encapsulated `gon` to be used only when `notarytool` is not available (Xcode version < 13.0)
- Added `notarytool` pkg submission by using user, pass, teamId combination.

### How to test this code?

There is a cloned job in the ci-release named [testing-new-osx-notarization-ojs+release](https://ci-release.nodejs.org/job/testing-new-osx-notarization-ojs+release/) where you can build with parameters as always. Please always use `disttype: test`.

As an example:

The Build 14 is passing. [see](https://ci-release.nodejs.org/job/testing-new-osx-notarization-ojs+release/nodes=osx11-release-pkg/14/). The `pkg` files are working fine 

<img width="623" alt="Screenshot 2023-09-15 at 20 37 15" src="https://github.com/nodejs/node/assets/5110813/8670e5d4-e29d-4369-9f63-27e6d30179e4">


The output from the Jenkins process is the expected:

![Screenshot 2023-09-15 at 20 36 26](https://github.com/nodejs/node/assets/5110813/5f90c5ee-816c-4783-9c9b-7a349f3d8722)


### Notes

In order to properly work is required to include a new environmental variable `NOTARIZATION_TEAM_ID` in the Jenkins job, this data is not sensitive information.

I am a bit rusty in bash, so feel free to improve the script. Only the linter steps are relevant in the CI validation for this PR. 

This is working with the existing credentials (`NOTARIZATION_PASSWORD` and `NOTARIZATION_ID`) so there is no breaking changes in the credentials for `gon`

### Context

**TL;DR:**

By November 1, 2023, we should be able to use `notarytool` to notarize Node.js. Currently, we use [`gon`](https://github.com/mitchellh/gon), but there is no support for `notarytool` yet. Additionally, we want to replace [`gon`](https://github.com/mitchellh/gon) as `notarytool` currently provides the option to wait for the notarization to be completed with the argument `--wait`.

- Fix https://github.com/nodejs/build/issues/3385.
- CC: @nodejs/build
